src/offsetPath.js

```diff
@@ -2,30 +2,32 @@
 'use strict';
 
 (function () {
-  function basicShapeInset(arguments) {
-    // WIP
-    return null;
-  }
-
-  function basicShapeCircle(arguments) {
-    var isNumeric = internalScope.isNumeric;
-    var argumentList = arguments.split(/\s+/);
-
-    var radius = argumentList[0];
-    // Negative radius is invalid
-    // Radius defaults to closest side 
-
-    return null;
-  }
-
-  function basicShapeEllipse(arguments) {
-    // WIP
-    return null;
-  }
-
-  function basicShapePolygon(arguments) {
-    // WIP
-    return null;
+  function basicShapePolygon (input) {
+    // TODO: Support the fill-rule option and %
+    var argumentList = input.split(',');
+    var coordinate = null;
+    var x = null;
+    var y = null;
+    var previousX = 0;
+    var previousY = 0;
+    var path = 'm 0 0';
+    // Do something here if not at least 3 vertices?
+    for (var i = 0; i < argumentList.length; i++) {
+      coordinate = argumentList[i].trim().split(/\s+/);
+      if (coordinate.length !== 2) {
+        return undefined;
+      }
+      x = internalScope.offsetDistanceParse(coordinate[0]);
+      y = internalScope.offsetDistanceParse(coordinate[1]);
+      if (!x || !y || x.unit === '%' || y.unit === '%') {
+        return undefined;
+      }
+      path += ' h ' + (x.value - previousX) + ' v ' + (y.value - previousY);
```

Comment:
path += ' l ' + (x.value - previousX) + ' ' + (y.value - previousY);

h 3 v 4 moves horizontally and then vertically, giving a path length of 7.

l 3 4 moves diagonally, giving a path length of 5